### PR TITLE
54 smooth lineages plot

### PR DESCRIPTION
**Pull request by priesgo**

### CODE REVIEW

File: covigator/dashboard/figures/lineages.py [hunk 1]
```diff
@@ -17,15 +17,16 @@
...
+    def get_lineages_plot(self, data_source: str, countries=None, lineages=None, time_period=14):
         logger.debug("Getting data on samples by country...")
         data = self.queries.get_accumulated_lineages_by_country(
             data_source=data_source, countries=countries, lineages=lineages)
         graph = dcc.Markdown("""**No data for the current selection**""")
         if data is not None and data.shape[0] > 0:
             logger.debug("Prepare plot on samples by lineage...")
             lineages = list(data.sort_values("cumsum", ascending=False).lineage.unique())
-
+            data.set_index('date', drop=False)
```
Comment (priesgo) [inline covigator/dashboard/figures/lineages.py:28]: I think this line is not doing anything, as `set_index()` does not affect the DataFrame on which it is called unless `inplace=True` is passed.

File: covigator/dashboard/figures/lineages.py [hunk 2]
```diff
@@ -36,8 +37,20 @@ def get_lineages_plot(self, data_source: str, countries=None, lineages=None):
...
+            # If time_period is False the un-smoothed data is plotted
+            smooth_data = data[['date', 'lineage', 'ratio_per_date', 'cumsum', 'count']]
+            if time_period:
+                # Group data by lineage to calculate correct average for each lineage in df
+                sma_df = smooth_data.groupby('lineage')[['lineage', 'ratio_per_date', 'cumsum', 'count']]
+                sma_df = sma_df.rolling(time_period, min_periods=1).mean()
+                sma_df = sma_df.reset_index(level='lineage')[['ratio_per_date', 'cumsum', 'count']]
+                # Update columns with moving average over selected period
+                smooth_data.loc[:, 'ratio_per_date'] = sma_df['ratio_per_date']
+                smooth_data.loc[:, 'cumsum'] = sma_df['cumsum']
```
Comment (priesgo) [inline covigator/dashboard/figures/lineages.py:50]: I was wondering if we want to do the rolling average also on `cumsum` and `count` considering that these values are only shown in the hover data, hence they do not affect the looks of the plot. What do you think? I don't have a strong preference here, just wondering.

File: covigator/dashboard/tabs/acknowledgements.py [hunk 3]
```diff
@@ -25,8 +25,8 @@ def get_tab_acknowledgements():
                 ],
                     style={"font-style": "italic", "margin-left": "50px"}),
                 html.P(
-                    html.A("https://doi.org/10.1101/2021.02.04.429765",
-                           href="https://doi.org/10.1101/2021.02.04.429765", target="_blank"),
+                    html.A("https://doi.org/10.1371/journal.pone.0249254",
+                           href="https://doi.org/10.1371/journal.pone.0249254", target="_blank"),
```
Comment (priesgo) [inline covigator/dashboard/tabs/acknowledgements.py:29]: Nice one! :) thanks

File: covigator/dashboard/tabs/lineages.py [hunk 4]
```diff
@@ -84,6 +85,14 @@ def get_lineages_tab_left_bar(queries: Queries, data_source: DataSource):
                 multi=True
             ),
             html.Br(),
+            dcc.Markdown("""Select time period for smoothing"""),
+            dcc.Dropdown(
+                id=ID_DROPDOWN_PERIOD,
+                options=[{'label': c, 'value': c} for c in range(1, 32)] + [{'label': 'Disable', 'value': False}],
```
Comment (priesgo) [inline covigator/dashboard/tabs/lineages.py:91]: I suggest you set the label to `"{} days".format(c)` so it is more explicit what the numbers correspond to.

**Review by priesgo — COMMENTED** [review]
Well done, I only have some minor comments. Once you look into those this looks ready to be merged.

File: covigator/dashboard/figures/lineages.py (hunk 1 above)
Comment (johausmann) [reply covigator/dashboard/figures/lineages.py:28]: Good hint. I thought I had already removed the line

**Pull request merged**